### DISCUSSION TITLE
Solid Walls - Two approaches?

### DISCUSSION THREAD

**Discussion by CbsTau** [ProjectPhysX/FluidX3D#240] [Q&A] [answered]
Hello,

I am modelling the flow generated by a rotating fan through a rectangular duct, and so I make the Side walls, and the Top and Bottom walls: TYPE_S

The problem then is that I cannot see into the box in shaded mode (VIS_FLAG_SURFACE).

Would it be the same numerically to model the solid walls of the ducting as follows:

		if(x==0u||x==Nx-1u||z==0u||z==Nz-1u) lbm.flags[n] = TYPE_E;   // Top and Side Walls set to Environment
		if(x==0u||x==Nx-1u||z==0u||z==Nz-1u) lbm.u[n] = 0.0f;             // Top and Side Walls velocity in x,y,z set to zero

This way I can see into the box to view the fan in shaded mode.

Is this approach equivalent numerically to setting the walls as TYPE_S?

Thanks in Advance!

![image](https://github.com/user-attachments/assets/c36131dc-0711-4ca0-b3dd-5477b673a5f0)

**Comment by ProjectPhysX** [comment, marked as answer]
Hi @CbsTau,

(`TYPE_S`) and (`TYPE_E` with `u=0`) are not equivalent. Use `TYPE_S` here, that ensures mass conservation.
To be able to look into the box in `VIS_FLAG_SURFACE` mode, modify [this line](https://github.com/ProjectPhysX/FluidX3D/blob/master/src/kernel.cpp#L2452) to
```c
	if(xyz.x<=0u||xyz.y<=0u||xyz.z<=0u||xyz.x>=def_Nx-2u||xyz.y>=def_Ny-2u||xyz.z>=def_Nz-2u) return;
```

Kind regards,
Moritz

**Reply by CbsTau** [reply]
Hi Moritz,
Many thanks for this clarification and work-around for visualizing inside closed boxes.

Kind regards, Clinton
